test/test_jit_cuda_fuser.py

```diff
@@ -0,0 +1,104 @@
+    if(RUN_CUDA):
+        torch._C._jit_register_cuda_fuser()
+    try:
+        yield
+    finally:
+        if(RUN_CUDA):
+            torch._C._jit_clear_cuda_fuser()
+        torch._C._jit_override_can_fuse_on_cpu(old_cpu_fuse)
+        torch._C._jit_override_can_fuse_on_gpu(old_gpu_fuse)
+
+
+class TestCudaFuser(JitTestCase):
+    def _has_cuda_fusion_group(self, graph):
+        has_cuda_fusion_group = False
+        for n in graph.nodes():
+            if n.kind() == 'prim::CudaFusionGroup':
+                has_cuda_fusion_group = True
+        return has_cuda_fusion_group
+
+    @unittest.skipIf(not RUN_CUDA, "requires CUDA")
+    @unittest.skipIf(GRAPH_EXECUTOR != ProfilingMode.PROFILING, "Requires profiling node to run cuda fuser")
+    @skipIfRocm
+    def test_const(self):
+        def t(x, y):
+            o = x + y
+            o = o + 2.0
+            return o
+        t_jit = torch.jit.script(t)
+        x = torch.randn(4, 8, dtype=torch.float, device="cuda")
+        y = torch.randn(4, 8, dtype=torch.float, device="cuda")
+        jit_o = t_jit(x, y)
+        jit_o = t_jit(x, y)
+        o = t(x, y)
+        self.assertEqual(o, jit_o)
+        self.assertTrue(self._has_cuda_fusion_group(t_jit.graph_for(x, y)))
+
+    @unittest.skipIf(not RUN_CUDA, "requires CUDA")
+    @unittest.skipIf(GRAPH_EXECUTOR != ProfilingMode.PROFILING, "Requires profiling node to run cuda fuser")
+    @skipIfRocm
+    def test_chunk(self):
+        def t(x, y, z, q):
+            o = x + q
+            x0, x1 = torch.chunk(o, 2)
+            o = x0 + x1
+            o = o + y
+            o = o * z
+            o = torch.relu(o)
+            return o
+        t_jit = torch.jit.script(t)
+        x = torch.randn(4, 8, dtype=torch.float, device="cuda")
+        y = torch.randn(2, 8, dtype=torch.float, device="cuda")
+        z = torch.randn(2, 8, dtype=torch.float, device="cuda")
+        q = torch.randn(4, 8, dtype=torch.float, device="cuda")
+        jit_o = t_jit(x, y, z, q)
+        jit_o = t_jit(x, y, z, q)
+        o = t(x, y, z, q)
+        self.assertEqual(o, jit_o)
+        self.assertTrue(self._has_cuda_fusion_group(t_jit.graph_for(x, y, z, q)))
+
+    @unittest.skipIf(not RUN_CUDA, "requires CUDA")
+    @unittest.skipIf(GRAPH_EXECUTOR != ProfilingMode.PROFILING, "Requires profiling node to run cuda fuser")
+    @skipIfRocm
+    def test_scalar_input(self):
+        def t(x, y, z):
+            # type: (Tensor, Tensor, float) -> Tensor
+            o = x + y
+            o = o + z
+            return o
+        t_jit = torch.jit.script(t)
+        x = torch.randn(4, 8, dtype=torch.float, device="cuda")
+        y = torch.randn(4, 8, dtype=torch.float, device="cuda")
+        jit_o = t_jit(x, y, 2.0)
+        jit_o = t_jit(x, y, 2.0)
+        o = t(x, y, 2.0)
+        self.assertEqual(o, jit_o)
+        self.assertTrue(self._has_cuda_fusion_group(t_jit.graph_for(x, y, 2.0)))
+
+
+if __name__ == '__main__':
+    with enable_new_fuser():
```

Comment:
I suspect that this won't work with `pytest` (i.e. if one directly invokes this test with `pytest test/test_jit_cuda_fuser.py`). Could you please double-check that? A solution would be to define `setUp` and `tearDown` methods in `TestCudaFuser` and enable/disable the new fuser from there.